eth/tracers/firehose_concurrency.go

```diff
@@ -0,0 +1,103 @@
+package tracers
+
+import (
+	pbeth "github.com/streamingfast/firehose-ethereum/types/pb/sf/ethereum/type/v2"
+	"sync"
+)
+
+type blockPrintJob struct {
+	block    *pbeth.Block
+	finality *FinalityStatus
+}
+
+type outputJob struct {
+	blockNum uint64
+	data     []byte
+}
+
+type ConcurrentFlushQueue struct {
+	bufferSize int
+
+	startSignal    chan uint64
+	jobQueue       chan *blockPrintJob
+	outputQueue    chan *outputJob
+	printBlockFunc func(block *pbeth.Block, finality *FinalityStatus)
+	outputFunc     func([]byte)
+
+	jobWG     sync.WaitGroup
+	outputWG  sync.WaitGroup
+	closeOnce sync.Once
+}
+
+func NewConcurrentFlushQueue(bufferSize int, printBlockFunc func(*pbeth.Block, *FinalityStatus), outputFunc func([]byte)) *ConcurrentFlushQueue {
+	return &ConcurrentFlushQueue{
+		startSignal:    make(chan uint64, 1),
+		jobQueue:       make(chan *blockPrintJob, bufferSize),
+		outputQueue:    make(chan *outputJob, bufferSize),
```

Comment:
Buffer size should be the same as number of worker taken in start, might be good to merge `Start` input with this one, which would mean that we keep the `bufferSize` struct variable. Maybe with another name though.